### DISCUSSION TITLE
displaying video by link in react-markdown

### DISCUSSION THREAD

**Discussion by hellhorse123** [remarkjs/remark#957] [Q&A] [answered]
Please tell me if it is currently possible to display videos in react-markdown as images (without resorting to HTML tags like rehypeRaw).
Is there a code like this with pictures `![alt_text](https://example.com/image.jpg)` ?
It would be great to have this option in the future

**Comment by ChristianMurphy** [comment, marked as answer]
Yes, it is possible.
By using a plugin to modify the syntax tree (see https://unifiedjs.com/learn/ for more on this)
The way markdown is converted to HTML can be modified by setting `node.data.hName`, `node.data.hProperties`, and `node.data.hChildren` as documented in https://github.com/syntax-tree/mdast-util-to-hast#examples

If you are looking for some inspiration, the Gatsby community has made some similar plugins https://www.npmjs.com/search?q=remark%20video

**Reply by ChristianMurphy** [reply]
It could also be done by creating an `img` component in `react-markdown` (see https://github.com/remarkjs/react-markdown#appendix-b-components)
And checking the extension to determine if the link is to an image or video.
